### PR TITLE
SI-6394 fixes macros.Context.enclosingClass

### PR DESCRIPTION
**Pull request by xeno-by**
review @gkossakowski @adriaanm

### CODE REVIEW

File: src/compiler/scala/reflect/macros/runtime/Enclosures.scala [hunk 1]
```diff
@@ -14,7 +14,7 @@ trait Enclosures {
   // vals are eager to simplify debugging
   // after all we wouldn't save that much time by making them lazy
   val macroApplication: Tree                 = expandee
-  val enclosingClass: Tree                   = site.enclClass.tree
+  val enclosingClass: Tree                   = enclTrees.collect{ case x: ImplDef => x }.headOption getOrElse EmptyTree
```
Comment (gkossakowski) [inline src/compiler/scala/reflect/macros/runtime/Enclosures.scala]: I'm not sure if this is the best solution. I'll ask @adriaanm when I have a chance.

Comment (paulp) [inline src/compiler/scala/reflect/macros/runtime/Enclosures.scala]: Ignoring the tree logic, the line is suboptimal:

```
enclTrees collectFirst { case x: ImplDef => x } getOrElse EmptyTree
```

Comment (gkossakowski) [inline src/compiler/scala/reflect/macros/runtime/Enclosures.scala]: Very good point.

**Comment by xeno-by** [comment]
PLS REBUILD ALL

**Comment by paulp** [comment]
This looks good to me.

**Comment by gkossakowski** [comment]
Ok, merging it then.

**Pull request merged**